transcom/mymove
# B-18413 prime sim sit address change and dest sit form changes

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A882392&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

We are transitioning to the shipment address being used for the `final_destination_address` on the service item level. This means that we no longer need the components in the Prime Simulator handling a SIT Destination Address Change, as well as removing the address form when requesting destination SIT.

This PR does the following:
- removes address form when requesting destination SIT
- added in some information in the dest SIT form for clarity/understanding of functionality
- removed `Request Destination SIT Address Change` button in `MoveDetails.jsx` component
- removed route & associated components for requesting SIT destination address change
- added tests & test file for destination SIT form
- added in functionality of `createNonSITAddressUpdateRequest` endpoint with new components and form with validation
- added in `createNonSITAddressUpdateRequest` endpoint to `primeApi.js`
- tests for all new components

**NOTE** This PR is dependent on [this BL item](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A882395&RoomContext=TeamRoom%3A808731&concept=TeamRoom) as for functionality. Please ensure that this does not get merged prior to the above mentioned PR.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test Scenario 1

1. Access MM office and open prime simulator
2. Open any move
3. Confirm the `Request Destination SIT Address Change` button is no longer in shipment summaries
4. Click on `Add service item` button
5. In the `Service item type` dropdown, select `Destination SIT`
6. Confirm there are no address fields
7. Confirm that the info appears above `Create service item` button and is clear in understanding

### How to test Scenario 2

1. Access MM office and open prime simulator
2. Open any move
3. Confirm the `Update Shipment Destination Address` is  on the page
4. Click on da button
5. Change whatever values you want - confirm that form is disabled when `Address 1`, `City`, and `Contractor Remarks` are blank
6. Send it
7. Confirm the new address is updated on the page
8. If the ZIP code changed drastically and requires TOO approval, find the move as a TOO and confirm it requires review

## Screenshots
![Screenshot 2023-12-29 at 11 36 42 AM](https://github.com/transcom/mymove/assets/136510600/d44335c2-ca34-480f-b0ae-eb029a6bb105)

Destination SIT request form
![Screenshot 2023-12-29 at 11 37 11 AM](https://github.com/transcom/mymove/assets/136510600/5563b133-6cdb-4f30-b54c-c8c36ffc30da)

New button added to `Shipment.jsx` to call `createNonSITAddressUpdateRequest` endpoint
![Screenshot 2024-01-02 at 10 59 12 AM](https://github.com/transcom/mymove/assets/136510600/051f98de-d838-4a4a-9fca-6ce8becfb73c)

Form for `createNonSITAddressUpdateRequest` endpoint
![Screenshot 2024-01-02 at 11 00 32 AM](https://github.com/transcom/mymove/assets/136510600/428c3d08-a0df-4e17-b776-21e1515ec466)
